openmetadata-ui/src/main/resources/ui/src/pages/TagPage/TagPage.tsx

```diff
@@ -0,0 +1,305 @@
+import AssetsTabs, {
+  AssetsTabRef,
+} from '../../components/Glossary/GlossaryTerms/tabs/AssetsTabs.component';
+import PageLayoutV1 from '../../components/PageLayoutV1/PageLayoutV1';
+import { FQN_SEPARATOR_CHAR } from '../../constants/char.constants';
+import { EntityType } from '../../enums/entity.enum';
+import { SearchIndex } from '../../enums/search.enum';
+import { Tag } from '../../generated/entity/classification/tag';
+import { MOCK_GLOSSARY_NO_PERMISSIONS } from '../../mocks/Glossary.mock';
+import { searchData } from '../../rest/miscAPI';
+import { getTagByFqn } from '../../rest/tagAPI';
+import { getCountBadge } from '../../utils/CommonUtils';
+import { getEntityName } from '../../utils/EntityUtils';
+import Fqn from '../../utils/Fqn';
+import { getQueryFilterToExcludeTerm } from '../../utils/GlossaryUtils';
+import {
+  getClassificationDetailsPath,
+  getClassificationTagPath,
+} from '../../utils/RouterUtils';
+import {
+  escapeESReservedCharacters,
+  getEncodedFqn,
+} from '../../utils/StringsUtils';
+import './tagpage.less';
+
+const TagPage = () => {
+  const { t } = useTranslation();
+  const { pathname } = useLocation();
+  const history = useHistory();
+  const { tab } = useParams<{ tab: string }>();
+  const tagFqn = pathname.split('/').find((path) => path.includes('.'));
+  const [isLoading, setIsLoading] = useState(false);
+  const [tagItem, setTagItem] = useState<Tag>();
+  const [assetModalVisible, setAssetModalVisible] = useState(false);
+  const [assetCount, setAssetCount] = useState<number>(0);
+  const assetTabRef = useRef<AssetsTabRef>(null);
+  const [previewAsset, setPreviewAsset] =
+    useState<EntityDetailsObjectInterface>();
+  const [breadcrumb, setBreadcrumb] = useState<
+    TitleBreadcrumbProps['titleLinks']
+  >([]);
+
+  const activeTab = useMemo(() => {
+    return tab ?? 'overview';
+  }, [tab]);
+
+  const handleAssetClick = useCallback(
+    (asset?: EntityDetailsObjectInterface) => {
+      setPreviewAsset(asset);
+    },
+    []
+  );
+
+  const getTagData = async () => {
+    try {
+      setIsLoading(true);
+      if (tagFqn) {
+        const response = await getTagByFqn(tagFqn);
+        setTagItem(response);
+      }
+    } catch (e) {
+      // handle error
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const handleBreadcrumb = (fqn: string) => {
+    if (!fqn) {
+      return;
+    }
+
+    const arr = Fqn.split(fqn);
+    const dataFQN: Array<string> = [];
+    const newData = [
+      {
+        name: 'Classifications',
+        url: getClassificationDetailsPath(arr[0]),
+        activeTitle: false,
+      },
+      ...arr.slice(0, -1).map((d) => {
+        dataFQN.push(d);
+
+        return {
+          name: d,
+          url: getClassificationDetailsPath(dataFQN.join(FQN_SEPARATOR_CHAR)),
+          activeTitle: false,
+        };
+      }),
+    ];
+
+    setBreadcrumb(newData);
```

Comment:
We can move it to useMemo and consume it instead of a state variable

Reply:
Updated.